apispec/ext/marshmallow/swagger.py

```diff
@@ -62,7 +88,10 @@ def _observed_name(field, name):
 
 
 def _get_json_type_for_field(field):
-    json_type, fmt = FIELD_MAPPING.get(type(field), ('string', None))
+    if hasattr(type(field), CUSTOM_FIELD_MAPPING_ATTR):
```

Comment:
Minor: If you remove `type` here, this function will work for both type classes and type instances.